src/graphql/data/TopTokenQuery.ts

```diff
@@ -115,11 +117,16 @@ export function useTopTokenQuery(page: number) {
     }
   )
 
-  const topTokens: TokenData[] | undefined = topTokenData.topTokenProjects?.map((token) =>
-    token?.tokens?.[0].address
+  const topTokens: TokenData[] | undefined = topTokenData.topTokenProjects?.map((token) => {
+    let address = token?.tokens?.[0].address
```

Comment:
Can this instead be compared to existing constants? See src/constants/tokens.ts specifically nativeOnChain (note you can also get this as a hook through useNativeCurrency).

You can probably compare this with ExtendedEther.onChain(chain) to make it more cross-chain compatible.